sycl/source/detail/kernel_info.hpp

```diff
@@ -56,14 +56,63 @@ get_kernel_info(RT::PiKernel Kernel, const plugin &Plugin) {
 }
 
 // Device-specific methods
+template <typename Param>
+uint32_t get_kernel_device_specific_info_with_input(RT::PiKernel Kernel,
```

Comment:
Do we expect to throw this internal interface away once deprecated public one is removed? If so, we can add a comment/TODO about that.

Reply:
I have added a comment about its removal when the corresponding deprecated info query is removed.